includes/class-wc-stripe-api.php

```diff
@@ -281,7 +316,27 @@ public static function retrieve( $api ) {
 			// Stripe redacts API keys in the response.
 			WC_Stripe_Logger::log( "Error: GET {$api} returned a 401" );
 
+			++$invalid_api_keys_count;
+			WC_Stripe_Database_Cache::set( self::INVALID_API_KEYS_CACHE_KEY, $invalid_api_keys_count, self::INVALID_API_KEYS_DELAY_IN_SECONDS );
+
+			if ( $invalid_api_keys_count >= self::INVALID_API_KEYS_COUNT_THRESHOLD ) {
+				WC_Stripe_Logger::error(
+					'Invalid API keys request rate limit exceeded',
+					[
+						'count'      => $invalid_api_keys_count,
+						'next_retry' => date_i18n( 'Y-m-d H:i:sP', time() + self::INVALID_API_KEYS_DELAY_IN_SECONDS ),
+					]
+				);
+
+				// We need to invalidate the Account Data cache here, so that the UI shows the "Connect to Stripe" button.
+				WC_Stripe_Database_Cache::delete( WC_Stripe_Account::ACCOUNT_CACHE_KEY );
+			}
+
 			return null; // The UI expects this empty response in case of invalid API keys.
+
+		} elseif ( null !== $invalid_api_keys_count ) {
+			// We only track consecutive 401 errors, so we delete the cache when we get a valid response.
+			WC_Stripe_Database_Cache::delete( self::INVALID_API_KEYS_CACHE_KEY );
 		}
```

Comment:
I am not sure we even need the `elseif`. The code above will always `return null;`, so we can handle this via a code comment. (Non-blocking.)
```suggestion
		}

		// We got a valid, non-401 response, so clear the invalid API key count if it is present.
		if ( null !== $invalid_api_keys_count ) {
			WC_Stripe_Database_Cache::delete( self::INVALID_API_KEYS_CACHE_KEY );
		}
```

Reply:
That said, what about cases where we never reach Stripe at all due to networking issues or a local error? Should we not check the HTTP response codes before clearing the cache?